lib/linked_list.rb

```diff
@@ -1,45 +1,101 @@
 require_relative 'node'
+require "pry"
 
 class LinkedList
   attr_reader :head
 
   def initialize
     @head = nil
+    # why is head initially set to nil?
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1) because inserting one node will never scale. It will always just be creating one node. 
+  # Space complexity - O(1) because inserting one node will always use one location in memory
   def add_first(data)
-
+    new_node = Node.new(data)
+    new_node.next = @head
+    # Need to put the node to another node or else it'll just
+    # be a floating node
+    @head = new_node
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1) because searching for the head node node will never scale. 
+  # Space complexity - O(1) because searching for the first node will never scale.
+  # a little confused as to how the space complexity is 0(1) though. We aren't storing anything.
```

Comment:
Exactly the amount of space doesn't change when you call this method, it's "constant".